src/Auth0.OidcClient.Core/IAuth0Client.cs

```diff
@@ -1,66 +1,69 @@
-﻿using System.Collections.Generic;
-using System.Threading.Tasks;
+﻿using System.Threading.Tasks;
 using IdentityModel.OidcClient;
 using IdentityModel.OidcClient.Browser;
 using IdentityModel.OidcClient.Results;
 
 namespace Auth0.OidcClient
 {
+    /// <summary>
+    /// Interface for performing authentication and authorization operations with Auth0 using the
+    /// underlying <see cref="IdentityModel.OidcClient.OidcClient"/>.
+    /// </summary>
     public interface IAuth0Client
     {
         /// <summary>
         /// Launches a browser to log the user in.
         /// </summary>
         /// <param name="extraParameters">Any extra parameters that need to be passed to the authorization endpoint.</param>
-        /// <returns></returns>
+        /// <returns>A <see cref="LoginResult"/> containing the tokens and claims.</returns>
         Task<LoginResult> LoginAsync(object extraParameters = null);
 
         /// <summary>
-        /// Launches a browser to log the user out and clear the Auth0 SSO Cookie
+        /// Launches a browser to log the user out and clear the Auth0 SSO Cookie.
         /// </summary>
-        /// <returns></returns>
+        /// <returns>A <see cref="BrowserResultType"/> indicating whether the logout was successful.</returns>
         Task<BrowserResultType> LogoutAsync();
 
         /// <summary>
-        /// Launches a browser to log the user out and clear the Auth0 SSO Cookie
+        /// Launches a browser to log the user out and clear the Auth0 SSO Cookie.
         /// </summary>
-        /// <param name="federated">Indicates whether the user should also be logged out of their identity provider.</param>
-        /// <returns></returns>
+        /// <param name="federated">Wether to log the user out of their federated identity provider.</param>
+        /// <returns>A <see cref="BrowserResultType"/> indicating whether the logout was successful.</returns>
         Task<BrowserResultType> LogoutAsync(bool federated);
 
         /// <summary>
-        /// Generates an <see cref="IdentityModel.OidcClient.AuthorizeState"/> containing the URL, state, nonce and code challenge which can
+        /// Generates an <see cref="AuthorizeState"/> containing the URL, state, nonce and code challenge which can
         /// be used to redirect the user to the authorization URL, and subsequently process any response by calling
         /// the <see cref="ProcessResponseAsync"/> method.
         /// </summary>
-        /// <param name="extraParameters"></param>
-        /// <returns></returns>
+        /// <param name="extraParameters">Additional parameters to send to the login endpoint.</param>
+        /// <returns>A <see cref="AuthorizeState"/> with necessary URLs, nonce, state and code verifiers.</returns>
```

Comment:
> code verifiers

What are those?

Reply:
It returns one of these https://github.com/IdentityModel/IdentityModel.OidcClient/blob/master/src/Shared/AuthorizeState.cs